rpc: add cli examples, update docs

rkrux opened the pull request:
### add cli example for `walletcreatefundedpsbt` and `createpsbt` RPCs
The only example present earlier was one that creates an OP_RETURN output. This
    lack of examples has discouraged me earlier to use this RPC. Adding an example
    that creates PSBT sending bitcoin to address, a scenario that is much more common.

### rpc: update the doc for `data` field in `outputs` argument
It was not evident to me that this field creates an `OP_RETURN` output until
    I read the code and tried it out. Thus, making the doc explicitly mention it.
This affects docs of the following RPCs:
`bumpfee`, `psbtbumpfee`, `send`, `walletcreatefundedpsbt`, `createpsbt`,
and `createrawtransaction`

rkrux committed:
rpc: add cli example for `walletcreatefundedpsbt` RPC

The only example present earlier was one that creates an OP_RETURN output. This
lack of examples has discouraged me earlier to use this RPC. Adding an example
that creates PSBT sending bitcoin to address, a scenario that is much more common.

rkrux committed:
rpc: update the doc for `data` field in `outputs` argument

This affects docs of the following RPCs:
`bumpfee`, `psbtbumpfee`, `send`, `walletcreatefundedpsbt`, `createpsbt`,
and `createrawtransaction`

It was not evident to me that this field creates an `OP_RETURN` output until
I read the code and tried it out. Thus, making the doc explicitly mention it.

i-am-yuvi commented:
ACK 52c66d9166b97f0a375d3accdd94eb5e9154862f

luke-jr commented on src/wallet/rpc/spend.cpp:
```diff
@@ -1729,8 +1729,12 @@ RPCHelpMan walletcreatefundedpsbt()
                                 },
                                 RPCExamples{
-                            "\nCreate a transaction with no inputs\n"
+                            "\nCreate a PSBT with a specified input and an OP_RETURN output:\n"
```
We shouldn't be encouraging this in the first place. Suggest replacing the example with yours.

rkrux replied:
I tend to agree, updated.

rkrux commented:
Updated example output of `walletcreatefundedpsbt` RPC:

```
Examples:

Create a PSBT with automatically picked inputs that sends 0.5 BTC to an address and has a fee rate of 2 sat/vB:
> bitcoin-cli walletcreatefundedpsbt "[]" "[{\"bc1q09vm5lfy0j5reeulh4x5752q25uqqvz34hufdl\":0.5}]" 0 "{\"add_inputs\":true,\"fee_rate\":2}"

Create the same PSBT as the above one instead using named arguments:
> bitcoin-cli -named walletcreatefundedpsbt outputs="[{\"bc1q09vm5lfy0j5reeulh4x5752q25uqqvz34hufdl\":0.5}]" add_inputs=true fee_rate=2
```

1440000bytes reviewed (commented):
ACK https://github.com/bitcoin/bitcoin/pull/31958/commits/0ad066c85a46208041f816d4edba36e8712763e9

Related PR: https://github.com/bitcoin/bitcoin/pull/27454

Chand-ra commented:
Concept ACK [8134a6](https://github.com/bitcoin/bitcoin/pull/31958/commits/8134a6b5d40568dcf32fdb21163cb1792efddc27)

The proposed changes _do_ improve the readability of the documentation.

rkrux committed:
rpc: update RPC help of `createpsbt`

Update the example wherein the PSBT sends bitcoin to an address instead
of creating an OP_RETURN output. Also, update the RPC description to
reflect the fact that the created transaction is unsigned.

ryanofsky reviewed (commented):
Concept ACK 32dcec269bf33f7be28245d88a1d8f2889cc39ae. These seem like helpful clarifications, but I did not look into the details

musaHaruna reviewed (commented):
ACK [931117a](https://github.com/bitcoin/bitcoin/pull/31958/commits/931117a46f5854d487e13b2b1446b621409c8371)

I will test and try it out soon, But from what I understand on a conceptual level if am not wrong,

All the mentioned RPCs—bumpfee, psbtbumpfee, send, walletcreatefundedpsbt, createpsbt, and createrawtransaction—share a common behavior for the outputs. 
i.e Whenever the outputs object includes a key "data", the RPC will treat its value as hex-encoded data for an OP_RETURN output. That how those mentioned RPC's (docs) are affected as mentioned in the PR's description. Please correct me if am wrong.
